src/Aspire.Hosting/Dashboard/proto/Partials.cs

```diff
@@ -42,7 +53,9 @@ public static Resource FromSnapshot(ResourceSnapshot snapshot)
 
         foreach (var property in snapshot.Properties)
         {
-            resource.Properties.Add(new ResourceProperty { Name = property.Name, Value = property.Value });
+            var isSensitive = s_knownSensitiveResourcePropertyNames.Contains(property.Name);
+
+            resource.Properties.Add(new ResourceProperty { Name = property.Name, Value = property.Value, IsSensitive = isSensitive });
```

Comment:
One thing missing is the ability to set this on the CustomResourceSnapshot https://github.com/dotnet/aspire/blob/0490bcbc4318c22774d162b3f49382f148cfa6b2/src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs#L106. That would flow here instead of hard coding any known properties.

Reply:
It would be good to have a way to control what properties are sensitive from the app model, but that can happen in a follow up PR.

Reply:
Updated.

I originally wanted to create a separate schema that's sent by the resource service on connection, that defines things like known properties and how to handle them. Then we wouldn't end up merging and sending static reference data along with dynamic property data. But that's beyond the scope of this PR.

Reply:
We should make it a property (it's breaking change to make it required).

Reply:
Rolled back the API change.